app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt

```diff
@@ -482,21 +507,42 @@ class SettingsFragment : Fragment(R.layout.activity_settings_screen) {
             checkForDownload(isUserInitiated = true, isRetry = true)
         }
 
-        b.settingsActivityThemeRl.setOnClickListener{
+        b.settingsActivityThemeRl.setOnClickListener {
             b.settingsActivityThemeRl.isEnabled = false
             showDialogForTheme()
+            // the count-down-timer ticks once after 500ms, at the completion of which
+            // the settingsActivityThemeRl [is re-enabled again or add whatever is appropriate.
             object : CountDownTimer(500, 500) {
-                override fun onTick(millisUntilFinished: Long) {
-                }
+                override fun onTick(millisUntilFinished: Long) {}
 
                 override fun onFinish() {
-                    if(isAdded) {
+                    if (isAdded) {
```

Comment:
redundant `if` see above

Reply:
isAdded is part of Fragment. This is to check if the fragment is attached to the Activity. 
Also in onFinish(), the isEnabled should be true.

isAdded check is added because the UI element should not updated when the fragment is not added to the activity.